src/Network/Wai/Handler/Warp/Systemd.hs

```diff
@@ -219,8 +219,9 @@ runSystemdWarp saSettings settings app = do
 
   case maybeSocket of
     Just socket -> do
-      setNonBlockIfNeeded (fdSocket socket)
-      runSettingsSocket settings' socket app
+      withFdSocket socket $ \fd -> do
+        setNonBlockIfNeeded fd
+        runSettingsSocket settings' socket app
```

Comment:
I'm not sure if this should be dedented or not.

Reply:
It's only for ensuring a reference to `socket` to avoid running the finalizer too early. This was never an issue here, so this change is technically redundant (either way), but let's do it like this anyway to avoid `unsafe*` fatigue and to give the right example.